File: kotlin-spring/src/main/kotlin/traveltimeplatform/models/ResponseGeocodingProperties.kt
package traveltimeplatform.models

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import traveltimeplatform.models.ResponseMapInfoFeatures
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

/**
 * 
 * @param name 
 * @param label 
 * @param score 
 * @param houseNumber 
 * @param street 
 * @param region 
 * @param regionCode 
 * @param neighbourhood 
 * @param county 
 * @param macroregion 
 * @param city 
 * @param country 
 * @param countryCode 
 * @param continent 
 * @param postcode 
 * @param features 
 */
data class ResponseGeocodingProperties (

        @get:NotNull 
        @JsonProperty("name") val name: String,

        @get:NotNull 
        @JsonProperty("label") val label: String,

        @JsonProperty("score") val score: Double? = null,

        @JsonProperty("house_number") val houseNumber: String? = null,

        @JsonProperty("street") val street: String? = null,

        @JsonProperty("region") val region: String? = null,

        @JsonProperty("region_code") val regionCode: String? = null,

        @JsonProperty("neighbourhood") val neighbourhood: String? = null,

        @JsonProperty("county") val county: String? = null,

        @JsonProperty("macroregion") val macroregion: String? = null,

        @JsonProperty("city") val city: String? = null,

        @JsonProperty("country") val country: String? = null,

        @JsonProperty("country_code") val countryCode: String? = null,

        @JsonProperty("continent") val continent: String? = null,

        @JsonProperty("postcode") val postcode: String? = null,

        @JsonProperty("features") val features: ResponseMapInfoFeatures? = null
) {

}

File: php-ze-ph/src/App/DTO/ResponseGeocodingProperties.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 */
class ResponseGeocodingProperties
{
    /**
     * @DTA\Data(field="name")
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $name;
    /**
     * @DTA\Data(field="label")
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $label;
    /**
     * @DTA\Data(field="score", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"float"})
     * @var float
     */
    public $score;
    /**
     * @DTA\Data(field="house_number", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $house_number;
    /**
     * @DTA\Data(field="street", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $street;
    /**
     * @DTA\Data(field="region", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $region;
    /**
     * @DTA\Data(field="region_code", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $region_code;
    /**
     * @DTA\Data(field="neighbourhood", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $neighbourhood;
    /**
     * @DTA\Data(field="county", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $county;
    /**
     * @DTA\Data(field="macroregion", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $macroregion;
    /**
     * @DTA\Data(field="city", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $city;
    /**
     * @DTA\Data(field="country", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $country;
    /**
     * @DTA\Data(field="country_code", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $country_code;
    /**
     * @DTA\Data(field="continent", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $continent;
    /**
     * @DTA\Data(field="postcode", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $postcode;
    /**
     * @DTA\Data(field="features", nullable=true)
     * @DTA\Strategy(name="Object", options={"type":\App\DTO\ResponseMapInfoFeatures::class})
     * @DTA\Validator(name="Dictionary", options={"type":\App\DTO\ResponseMapInfoFeatures::class})
     * @var \App\DTO\ResponseMapInfoFeatures
     */
    public $features;
}
